crates/bevy_render/src/mesh/primitives/extrusion.rs

```diff
@@ -0,0 +1,348 @@
+use bevy_math::{
+    primitives::{Annulus, Capsule2d, Circle, Ellipse, Extrusion, Primitive2d},
+    Vec2, Vec3,
+};
+
+use crate::mesh::{Indices, Mesh, VertexAttributeValues};
+
+use super::{MeshBuilder, Meshable};
+
+/// An enum representing segments of the perimeter of extrudable meshes
+pub enum PerimeterSegment {
+    /// This segment of the perimeter will be shaded smooth.
+    ///
+    /// You may want to use this for curved segments
+    Smooth {
```

Comment:
I think it would also be good to say more about how the smooth normals are actually computed — it doesn't have to be that precise, but I found the parameters kind of confusing before reading through the source code to learn how they are used.